packages/components/src/autocomplete/control.js

```diff
@@ -0,0 +1,216 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, createRef } from '@wordpress/element';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Tags from './tags';
+
+/**
+ * A search control to allow user input to filter the options.
+ */
+class SearchControl extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			isActive: false,
+		};
+
+		this.input = createRef();
+
+		this.updateSearch = this.updateSearch.bind( this );
+		this.onFocus = this.onFocus.bind( this );
+		this.onBlur = this.onBlur.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+	}
+
+	updateSearch( onSearch ) {
+		return event => {
+			onSearch( event.target.value );
+		};
+	}
+
+	onFocus( onSearch ) {
+		return event => {
+			this.setState( { isActive: true } );
+			onSearch( event.target.value );
+		};
+	}
+
+	onBlur() {
+		this.setState( { isActive: false } );
+	}
+
+	onKeyDown( event ) {
+		const { selected, onChange, query } = this.props;
+
+		if ( 8 === event.keyCode && ! query && selected.length ) {
```

Comment:
<s>Minor: Can you add a comment that 8 is the keyCode for backspace?</s>

Can we use the `BACKSPACE` constant from `@wordpress/keycodes` here?